lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -138,8 +139,6 @@ def call(env)
           # rubocop:disable Metrics/MethodLength
           def set_request_tags!(trace, request_span, env, status, headers, response, original_env)
             request_header_collection = Header::RequestHeaderCollection.new(env)
```

Comment:
It looks like we could pass `request_header_collection`  to `HeaderTagging.tag_request_headers(request_span, env, configuration)`

`HeaderTagging.tag_request_headers(request_span, request_header_collection, configuration)`. Internally in `tag_request_headers` we use the `env` to construct a `Header::RequestHeaderCollection`. We can avoid the duplicate `Header::RequestHeaderCollection`

Reply:
I gave it a shot!
I had to make changes inside `HeaderTagging#tag_request_headers` to prevent it from double wrapping the argument, but I think it's a good change.